src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -48,24 +50,70 @@ internal class IssuesSnapshot : WpfTableEntriesSnapshotBase
         private readonly IList<IssueMarker> issueMarkers;
         private readonly IReadOnlyCollection<IssueMarker> readonlyIssueMarkers;
 
+        /// <summary>
+        /// Logical identifier for a set of issues produced by an analysis run.
+        /// Every snapshot that has the same analysis id describes the same set of
+        /// issues in the same order. The implementation of "IndexOf" depends on this.
+        /// </summary>
+        public Guid AnalysisRunId { get; }
+
         public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers;
 
-        internal IssuesSnapshot(string projectName, string filePath, int versionNumber, IEnumerable<IssueMarker> issueMarkers)
-            : this(projectName, filePath, versionNumber, issueMarkers, new AnalysisSeverityToVsSeverityConverter(), new RuleHelpLinkProvider())
+        // Every snapshot has a unique version number. It doesn't matter what it is, as
```

Comment:
The handling of version numbers doesn't need to be visible outside the class. We only need to ensure it's an incrementing number.